### ISSUE TITLE
groupBitmapAndState from distributed table

### ISSUE THREAD

**Issue by phhcer** [ClickHouse/ClickHouse#15931]
I have a distributed table like this and local table similar schema.
```
CREATE TABLE test_dis (
    ds String, 
    rbm AggregateFunction(groupBitmap, UInt32), 
    dim String
) ENGINE = Distributed(cluster, test, test, rand())
```
column  rbm is generated with this sql
```
select
    ds,
    groupBitmapState(id) as rbm,
    dim
from a
group by 
    ds,
    dim
```
when I want to get the Intersection rbm from a distributed table like this
```
select
    groupBitmapAnd(rbm)
from test_dis
```
It raised Error:

> Code: 43. DB::Exception: Received from x.x.x.x:9000. DB::Exception: The type AggregateFunction(groupBitmap, UInt32) of argument for aggregate function groupBitmap is illegal, because it cannot be used in Bitmap operations: while receiving packet from x.x.x.x:9000: While executing SourceFromInputStream.

However, when I use subquery to pull all data from remote machine to local 

```
select
    groupBitmapAnd(rbm)
from (
    select
        rbm
    from
        test_dis
)
```
It works! Is there any solution to avoid pull all data from remote machine by using subquery ?

**Comment by den-crane** [comment]
Not sure. 
But I think you should use in both cases

select
    **groupBitmapAndMerge**(rbm)
from test_dis


select
    **groupBitmapAndMerge**(rbm)
from (
    select
        rbm
    from
        test_dis
)

**Comment by phhcer** [comment]
> Not sure.
> But I think you should use -Merge in both cases
> 
> select **groupBitmapAndMerge**(rbm) from test_dis
> 
> select **groupBitmapAndMerge**(rbm) from (
  
  
**groupBitmapAndMerge** In both cases go wrong
  
  
> Code: 43. DB::Exception: Received from 9.221.46.144:9000. DB::Exception: Illegal type UInt32 of argument for aggregate function groupBitmapAnd.

But **groupBitmapMerge** success as usual. Shouldn’t I get intersection set for **AggregateFunction(groupBitmap, UInt32)** data?

I guess that subquery will trigger the merge mechanism so that I can use **groupBitmapAnd** from subquery?

**Comment by readme2gh** [comment]
I also meet the same problem.

**Comment by MaxWk** [comment]
+1 i have the same issue

**Comment by readme2gh** [comment]
> +1 i have the same issue

Hi,how is it going ?

**Comment by juntaozhang** [comment]
+1 i have the same issue

**Comment by villins** [comment]
Can try ```groupBitmapMergeStateOrDefault```

**Comment by abel-cheng** [comment]
Solved in [#26557](https://github.com/ClickHouse/ClickHouse/pull/26557)

**Comment by melvynator** [comment]
Closing given the problem seems to have been addressed. Thanks @abel-cheng
